src/job_cache/job_cache_impl_common.cpp

```diff
@@ -0,0 +1,251 @@
+/*
```

Comment:
everything but the remove_backing_files functions is old code that was just moved here so that it would have a common location across job_cache